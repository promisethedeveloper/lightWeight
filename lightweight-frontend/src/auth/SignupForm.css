.SignupForm-heading {
	color: #461111;
}

label {
	color: #a13333;
}

.SignupForm-button {
	color: white;
	background-color: #461111;
}

button:hover {
	background-color: #a13333;
	color: white;
}
