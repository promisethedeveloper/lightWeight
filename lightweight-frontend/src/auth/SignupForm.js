import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Alert from "../common/Alert";
import "./SignupForm.css";

const SignupForm = ({ signup }) => {
	const history = useHistory();
	const [formData, setFormData] = useState({
		username: "",
		password: "",
		firstName: "",
		lastName: "",
		email: "",
	});

	const [formErrors, setFormErrors] = useState([]);

	/**
	 * Handle form submit:
	 *
	 * Calls login function prop and if successful, redirect to "/"
	 */

	const handleSubmit = async (e) => {
		e.preventDefault();
		let result = await signup(formData);
		if (result.success) {
			history.push("/");
		} else {
			setFormErrors(result.errors);
		}
	};

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((data) => ({ ...data, [name]: value }));
	};

	return (
		<div className="SignupForm">
			<div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
				<h2 className="mb-3 SignupForm-heading">Sign Up</h2>
				<div className="card">
					<div className="card-body">
						<form onSubmit={handleSubmit}>
							<div className="form-group">
								<label htmlFor="username">Username</label>
								<input
									id="username"
									name="username"
									className="form-control"
									value={formData.username}
									onChange={handleChange}
								/>
							</div>
							<div className="form-group">
								<label htmlFor="password">Password</label>
								<input
									id="password"
									type="password"
									name="password"
									className="form-control"
									value={formData.password}
									onChange={handleChange}
								/>
							</div>
							<div className="form-group">
								<label htmlFor="firstName">First name</label>
								<input
									id="firstName"
									name="firstName"
									className="form-control"
									value={formData.firstName}
									onChange={handleChange}
								/>
							</div>
							<div className="form-group">
								<label htmlFor="lastName">Last name</label>
								<input
									id="lastName"
									name="lastName"
									className="form-control"
									value={formData.lastName}
									onChange={handleChange}
								/>
							</div>
							<div className="form-group">
								<label htmlFor="email">Email</label>
								<input
									id="email"
									name="email"
									type="email"
									className="form-control"
									value={formData.email}
									onChange={handleChange}
								/>
							</div>

							{formErrors.length ? (
								<Alert type="danger" messages={formErrors} />
							) : null}

							<button
								type="submit"
								className="btn float-right font-weight-bold SignupForm-button"
								onSubmit={handleSubmit}
							>
								SUBMIT
							</button>
						</form>
					</div>
				</div>
			</div>
		</div>
	);
};

export default SignupForm;
