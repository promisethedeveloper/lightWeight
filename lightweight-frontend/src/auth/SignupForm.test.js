import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SignupForm from "./SignupForm";

/** Smoke test */
it("renders without crashing", () => {
	render(
		<MemoryRouter>
			<SignupForm />
		</MemoryRouter>
	);
});

/** Snapshot test */
it("Should match snapshots", () => {
	const { asFragment } = render(<SignupForm />);
	expect(asFragment()).toMatchSnapshot();
});

it("renders a 'Sign Up' text", () => {
	const { getByText } = render(<SignupForm />);
	expect(getByText("Sign Up")).toBeInTheDocument();
});

it("renders form's label texts", () => {
	const { getByLabelText } = render(<SignupForm />);
	expect(getByLabelText("Username")).toBeInTheDocument();
	expect(getByLabelText("Password")).toBeInTheDocument();
	expect(getByLabelText("First name")).toBeInTheDocument();
	expect(getByLabelText("Last name")).toBeInTheDocument();
	expect(getByLabelText("Email")).toBeInTheDocument();
	expect(getByLabelText("Password")).toBeInTheDocument();
});
