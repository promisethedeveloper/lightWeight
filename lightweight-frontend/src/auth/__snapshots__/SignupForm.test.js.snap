// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="SignupForm"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h2
        class="mb-3 SignupForm-heading"
      >
        Sign Up
      </h2>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label
                for="username"
              >
                Username
              </label>
              <input
                class="form-control"
                id="username"
                name="username"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                class="form-control"
                id="password"
                name="password"
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="firstName"
              >
                First name
              </label>
              <input
                class="form-control"
                id="firstName"
                name="firstName"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="lastName"
              >
                Last name
              </label>
              <input
                class="form-control"
                id="lastName"
                name="lastName"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                class="form-control"
                id="email"
                name="email"
                type="email"
                value=""
              />
            </div>
            <button
              class="btn float-right font-weight-bold SignupForm-button"
              type="submit"
            >
              SUBMIT
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
